lib/nokogiri/xml/parse_options.rb

```diff
@@ -1,167 +1,406 @@
+    # - Nokogiri::XML::Schema.new (no block).
+    # - Nokogiri::XML::Schema.read_memory (no block).
+    # - XSD::XMLParser::Nokogiri.new (no block).
     #
-    # == Setting and unsetting parse options
+    # Certain other parsing methods use different options;
+    # see HTML5.
     #
-    # You can build your own combinations of parse options by using any of the following methods:
+    # ## Bitmap Constants
     #
-    # [ParseOptions method chaining]
+    # Each of the [parsing methods](rdoc-ref:ParseOptions@Parsing+Methods)
+    # discussed here accept an integer argument `options` that specifies parsing options.
     #
-    #   Every option has an equivalent method in lowercase. You can chain these methods together to
-    #   set various combinations.
+    # That integer value may be constructed using the bitmap constants defined in \ParseOptions.
     #
-    #     # Set the HUGE & PEDANTIC options
-    #     po = Nokogiri::XML::ParseOptions.new.huge.pedantic
-    #     doc = Nokogiri::XML::Document.parse(xml, nil, nil, po)
+    # Except for `STRICT` (see note below),
+    # each of the bitmap constants has a non-zero value
+    # that represents a bit in an integer value;
+    # to illustrate, here are a few of the constants, displayed in binary format (base 2):
     #
-    #   Every option has an equivalent <code>no{option}</code> method in lowercase. You can call these
-    #   methods on an instance of ParseOptions to unset the option.
+    # ```
+    # ParseOptions::RECOVER.to_s(2)  # => "1"
+    # ParseOptions::NOENT.to_s(2)    # => "10"
+    # ParseOptions::DTDLOAD.to_s(2)  # => "100"
+    # ParseOptions::DTDATTR.to_s(2)  # => "1000"
+    # ParseOptions::DTDVALID.to_s(2) # => "10000"
+    # ```
     #
-    #     # Set the HUGE & PEDANTIC options
-    #     po = Nokogiri::XML::ParseOptions.new.huge.pedantic
+    # Any of these constants may be used alone to specify a single option:
     #
-    #     # later we want to modify the options
-    #     po.nohuge # Unset the HUGE option
-    #     po.nopedantic # Unset the PEDANTIC option
+    # ```
+    # ParseOptions.new(ParseOptions::DTDLOAD)
+    # # => #<Nokogiri::XML::ParseOptions: ... strict, dtdload>
+    # ParseOptions.new(ParseOptions::DTDATTR)
+    # # => #<Nokogiri::XML::ParseOptions: ... strict, dtdattr>
+    # ```
     #
-    #   💡 Note that some options begin with "no" leading to the logical but perhaps unintuitive
-    #   double negative:
+    # Multiple constants may be ORed together to specify multiple options:
     #
-    #     po.nocdata # Set the NOCDATA parse option
-    #     po.nonocdata # Unset the NOCDATA parse option
+    # ```
+    # options = ParseOptions::BIG_LINES | ParseOptions::COMPACT | ParseOptions::NOCDATA
+    # ParseOptions.new(options)
+    # # => #<Nokogiri::XML::ParseOptions: ... strict, nocdata, compact, big_lines>
+    # ```
     #
-    #   💡 Note that negation is not available for STRICT, which is itself a negation of all other
```

Comment:
I think we need to keep this note somewhere, doesn't need to be a separate call-out though.

Reply:
We have this:

    #   Note that there is no unsetter `nostrict`,
    #   but the setter `recover` serves the same purpose:
    #

Reply:
Restored.